platform/Stream.cpp

```diff
@@ -112,14 +103,14 @@ void Stream::rewind() {
 }
 
 int Stream::isatty() {
-    return 0;
+    return 1;
```

Comment:
No, isatty() is public API of FileHandle and Stream, it does affect behaviour of the C library, and it's possible some third parties may be looking at it.

I agree changing that seems unlikely to break stuff, but changing it doesn't really fit with the fact we're planning to deprecate Stream because we think it's broken and isn't safe to change. I imagine fewer people are relying on the broken isatty than the broken read/write, but still. If we're not trying to fix those, not sure we should try to fix isatty.

There's a pretty big conceptual potential-breakage difference between a change that doesn't change Stream+Serial's existing APIs at all versus one that tweaks existing APIs a bit.

If, as I think, the isatty() change is the only API change to Stream in the PR, I'd be inclined to drop it to leave Stream untouched (some code does move to fdopen, but its content doesn't change).